src/EnergyPlus/UnitarySystem.cc

```diff
@@ -10891,9 +10923,12 @@ namespace UnitarySystems {
                 Fans::SimulateFanComponents(state, blankString, FirstHVACIteration, this->m_FanIndex, state.dataUnitarySystems->FanSpeedRatio);
             }
         }
-
-        Real64 SuppPLR = this->m_SuppHeatPartLoadFrac;
         if (this->m_SuppCoilExists) {
+            Real64 SuppPLR = 0.0;
+            if (this->m_NumOfSpeedSuppHeating > 0) {
+                this->calcMultiStageSuppCoilStageByLoad(state, SuppCoilLoad, FirstHVACIteration);
+            }
+            SuppPLR = this->m_SuppHeatPartLoadFrac;
             this->calcUnitarySuppHeatingSystem(state, FirstHVACIteration, SuppPLR, SuppCoilLoad);
             if ((state.dataLoopNodes->Node(this->m_SuppCoilAirOutletNode).Temp > this->DesignMaxOutletTemp) && SuppPLR > 0.0 &&
                 !this->m_SimASHRAEModel) {
```

Comment:
@yzhou601 did you ever look at what happens when the supp heater outlet temp exceeds the DesignMaxOutletTemp? Could you test this by entering a DesignMaxOutletTemp a little lower than what this heating coil outlet temp is to see if the outlet temp gets clipped correctly?

Reply:
I will add a test a scenario for this to my unit test